### PR TITLE
Update auth login cookie path

### PR DESCRIPTION
**Pull request by sjezewski**
And update tests

### CODE REVIEW

File: src/server/pfs/server/http.go [hunk 1]
```diff
@@ -107,27 +114,54 @@ type loginRequestPayload struct {
...
 }
+
+type notFoundRouter struct {
+}
+
+func (s notFoundRouter) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Header().Add("Content-Type", "text/html; charset=utf-8")
+	w.WriteHeader(http.StatusNotFound)
+	b := strings.NewReader("route not found")
+	fw := flushWriter{w: w}
```
Comment (msteffen) [inline src/server/pfs/server/http.go]: Per discussion: you wanted to limit the use of flushWriter to GetFile (http.ResponseWriter normally buffers the response body in memory, which doesn't work for GetFile where the body might be very large). It's not really necessary in calls that only return a tiny amount of data)

File: src/server/pfs/server/http.go [hunk 2]
```diff
@@ -107,27 +114,54 @@ type loginRequestPayload struct {
...
-	if data.Token == "" {
+	token := r.FormValue("Token")
+	if token == "" {
 		// Return 500
 		http.Error(w, "empty token provided", http.StatusInternalServerError)
 		return
 	}
-	w.Header().Add("Set-Cookie", fmt.Sprintf("%v=%v", auth.ContextTokenKey,
-		data.Token))
+	w.Header().Add("Set-Cookie", fmt.Sprintf("%v=%v;path=/", auth.ContextTokenKey,
```
Comment (msteffen) [inline src/server/pfs/server/http.go]: Note for posterity: Cookies set by `/auth/login` don't apply to `/pfs/...` unless you set `path=/` in the response

File: src/server/pfs/server/http.go [hunk 3]
```diff
@@ -107,27 +114,44 @@ type loginRequestPayload struct {
...
 	w.Header().Add("Access-Control-Allow-Origin", "*")
 	w.WriteHeader(http.StatusOK)
 }
+
+type notFoundRouter struct {
+}
+
+func (s notFoundRouter) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Header().Add("Content-Type", "text/html; charset=utf-8")
+	w.WriteHeader(http.StatusNotFound)
```
Comment (msteffen) [inline src/server/pfs/server/http.go]: One more nit: could you use `http.Error(w, "route not found", http.StatusNotFound)` here?

Comment (sjezewski) [reply src/server/pfs/server/http.go]: Yup, good call

**Pull request merged**